spec/models/provider_spec.rb

```diff
@@ -28,14 +28,29 @@
 
 require 'rails_helper'
 
-RSpec.describe Provider, type: :model do
+describe Provider, type: :model do
   subject { create(:provider) }
 
   describe 'associations' do
     it { should have_many(:sites) }
     it { should have_many(:users).through(:organisations) }
   end
 
+  describe 'changed_at' do
+    it 'is set on create' do
+      provider = Provider.create
+      expect(provider.changed_at).to be_present
+      expect(provider.changed_at).to eq provider.updated_at
+    end
+
+    it 'is set on update' do
+      provider = create(:provider, updated_at: 1.hour.ago)
+      provider.touch
+      expect(provider.changed_at).to eq provider.updated_at
+      expect(provider.changed_at).not_to be_within(1.second).of(1.hour.ago)
```

Comment:
timecop?

Reply:
Please expand. :)

Reply:
guessing this is the actual test subject

```ruby
provider.touch
``` inferring this

```ruby
 def touch_provider
    provider.update_changed_at
  end
```

hence the items in question


```ruby
expect(provider.changed_at).to eq provider.updated_at
expect(provider.changed_at).not_to be_within(1.second).of(1.hour.ago)
```

so this should be 
```ruby
expect(provider.updated_at).to be eq Time.now.utc
expect(provider.changed_at).to be eq Time.now.utc
```


therefore `TimeCop`

Reply:
Sure, we can do that.